R/graded.R

```diff
@@ -1,27 +1,123 @@
-#' Graded object for submission value
+#' Signal a final grade for a student's submission
 #'
-#' The return value from `graded` should be returned by every
-#' `*-check` chunk when used with [grade_learnr()].
+#' `graded()` is used to signal a final grade for a submission. In general, it
+#' is most helpful when used in [grade_this()], but it is used internally by
+#' other grading functions to signal that the student's submission has been
+#' graded. If you're using [grade_this()], you'll most likely want to use the
+#' helper functions `pass()`, `fail()`, `pass_if_equal()`, and
+#' `fail_if_equal()`, rather than to call `graded()` directly.
 #'
-#' `graded()` objects are signaled to the calling functions.
+#' @section Usage in `grade_learnr()`: 
+#'   
+#'   If \pkg{gradethis} is used in a [learnr::tutorial()] with the default
+#'   [gradethis_setup()], [grade_learnr()] expects the `*-check` chunk for an
+#'   exercise to return a function. When the exercise submission is to be
+#'   graded, [grade_learnr()] will call the checking function, providing it with
+#'   a consistent exercise submission environment — see [mock_this_exercise()]
+#'   for examples of this environment. The goal of this function is to evaluate
+#'   the submission and to return a final grade via `graded()`.
 #'
-#'   * [grade_result()] ignores when grades are created. `graded()` objects must be returned
-#'   * [grade_code()] handles `graded()` objects internally
-#'   * [grade_this()] will stop execution once a `pass()`, `pass_if_equal()`,
-#'     `fail()`, `fail_if_equal()`, or `grade()` is called. To generate programmatic grades,
-#'     use `graded()` or if statements around `pass*()` and `fail*()`
+#'   In general, tutorial authors will primarily use `graded()` and its helper
+#'   functions only when using [grade_this()]. Whenever one of these functions
+#'   is called inside [grade_this()], the submission checking will stop
+#'   immediately and the appropriate grade and feedback will be returned.
+#'
+#'   Internally, [grade_this_code()], [grade_result()], and [grade_code()] all
```

Comment:
Should we only mention `grade_this()` and `grade_this_code()`? (As `grade_result()` and `grade_code()` have been soft deprecated

Reply:
I guess this depends on how much we want to push people away from `grade_result()` and `grade_code()`. OTOH, I wanted to make it clear that these functions _shouldn't_ be used in any of the other grading functions. I suppose we can change this to something along the lines of "Internally other grading functions use `graded()`, so you should only uses the `graded()` functions with `grade_this()`."

Reply:
I see and understand now that now. Skimming failed me.